src/devices/Ili934x/Rgb565.cs

```diff
@@ -0,0 +1,199 @@
+                int lowByte = _value & 0xF8;
+                if (lowByte == 0)
+                {
+                    return 0;
+                }
+
+                return lowByte | 0x7;
+            }
+        }
+
+        public int G
+        {
+            get
+            {
+                int gbyte = ((Swap(_value) & 0x7E0) >> 3);
+                if (gbyte == 0)
+                {
+                    return 0;
+                }
+
+                return gbyte | 0x7;
+            }
+        }
+
+        public int B
+        {
+            get
+            {
+                int bbyte = (_value >> 5) & 0xF8;
+                if (bbyte == 0)
+                {
+                    return 0;
+                }
+
+                return bbyte | 0x7;
+            }
+        }
+
+        private void InitFrom(ushort r, ushort g, ushort b)
+        {
+            // get the top 5 MSB of the blue or red value
+            UInt16 retval = (UInt16)(r >> 3);
+            // shift right to make room for the green Value
+            retval <<= 6;
+            // combine with the 6 MSB if the green value
+            retval |= (UInt16)(g >> 2);
+            // shift right to make room for the red or blue Value
+            retval <<= 5;
+            // combine with the 6 MSB if the red or blue value
+            retval |= (UInt16)(b >> 3);
+
+            _value = Swap(retval);
+        }
+
+        /// <summary>
+        /// Convert a color structure to a byte tuple representing the colour in 565 format.
+        /// </summary>
+        /// <param name="color">The color to be converted.</param>
+        /// <returns>
+        /// This method returns the low byte and the high byte of the 16bit value representing RGB565 or BGR565 value
+        ///
+        /// byte    11111111 00000000
+        /// bit     76543210 76543210
+        ///
+        /// For ColorSequence.RGB (inversed!, the LSB is the top byte)
+        ///         GGGBBBBB RRRRRGGG
+        ///         43210543 21043210
+        /// </returns>
+        public static Rgb565 FromRgba32(Color color)
+        {
+            // get the top 5 MSB of the blue or red value
+            UInt16 retval = (UInt16)(color.R >> 3);
+            // shift right to make room for the green Value
+            retval <<= 6;
+            // combine with the 6 MSB if the green value
+            retval |= (UInt16)(color.G >> 2);
+            // shift right to make room for the red or blue Value
+            retval <<= 5;
+            // combine with the 6 MSB if the red or blue value
+            retval |= (UInt16)(color.B >> 3);
+
+            return new Rgb565(Swap(retval));
+        }
+
+        private static ushort Swap(ushort val) => (ushort)(val >> 8 | val << 8);
+
+        public ushort PackedValue
+        {
+            get
+            {
+                return _value;
+            }
+            set
+            {
+                _value = value;
+            }
+        }
```

Comment:
```suggestion
        public ushort PackedValue { get; set; }
```

and remove extra field

Reply:
In structs, I prefer to use explicit fields, to clearly see the size of the struct and also the order of the fields, since this is important for many use cases (here as well).